src/mainClass.js

```diff
@@ -0,0 +1,37 @@
+import { getData, getCategories, getAllLinks } from "./source";
+
+export class AllItems {
+  constructor(searchCategory, displayCategoryLink, transform, searchLinks) {
+    this.allCategories = searchCategory; // searchCategory()
```

Comment:
still confusing parameters searchCategory? allCategories. Call it onCategriesReceived, on both side class and UI
same for links.